File: docs/source/tutorial/code-snippets/painter-setup.cpp
#include <viren2d/viren2d.h>

auto painter = viren2d::CreatePainter();

// Either create an empty canvas (H=600, W=800):
painter->SetCanvas(600, 800, viren2d::Color(viren2d::NamedColor::NavyBlue));

// Initialize with an existing viren2d::ImageBuffer:
painter->SetCanvas(image_buffer);

// Or initialize from an image file:
painter->SetCanvas("path/to/image.jpg");

File: docs/source/tutorial/code-snippets/painter-setup.py
import viren2d

painter = viren2d.Painter()

# Either create an empty canvas:
painter.set_canvas_rgb(height=600, width=800, color='navy-blue')

# Initialize with a given NumPy ndarray `np_image`:
painter.set_canvas_image(image_numpy)

# Or initialize from an image file:
painter.set_canvas_filename('path/to/image.jpg')
